### PR TITLE
[docs] Scheduler features

### PR DESCRIPTION
**Pull request by stevhliu**
Continuation of #7817 (see comment [here](https://github.com/huggingface/diffusers/pull/7817#pullrequestreview-2048708715)) that refactors scheduler features for inference to their own doc. It includes:

- custom `timesteps` and `sigmas` showcasing AYS
- Karras sigmas
- `rescale_betas_zero_snr`

### CODE REVIEW

**Review by yiyixuxu — APPROVED** [review]
thanks!

**Comment by yiyixuxu** [comment]
cc @Beinsezii here
we are adding a general doc page for the scheduler now; anything else we should add here?

**Comment by Beinsezii** [comment]
I feel like a section on the timestep spacings would be beneficial, especially since they're related part of the [same paper referenced](https://hf.co/papers/2305.08891). The paper recommends `trailing` which is what I and a lot of others have settled on. `trailing` is unique in that its mutually exclusive with `steps_offset ≥ 1` as well.

**Comment by bghira** [comment]
a good demonstration of the current generation of models' two primary forms of residual noise would probably be a good idea though i can't think of how to integrate that. i just see it a lot and i think the community needs language to describe it with, and common solutions to try. probably for a separate doc

**Comment by stevhliu** [comment]
Thanks for the feedback, added a new section for timestep spacing! 

> a good demonstration of the current generation of models' two primary forms of residual noise

Good idea, maybe we can explore this in a separate PR :)

**Comment by Beinsezii** [comment]
Maybe a "Generation Quality" doc that has a bunch of common footguns. Like using Karras sigmas on models that weren't trained for it, or turning off `set_alpha_to_one`/`final_sigmas_type`.

Also I think solver order be explored in more depth either here or another doc because the best one is highly dependent on the rest of the params. Like, if you're going run 50 steps anyways a 1st order sampler will have plenty strong enough prediction with less hallucinations. Really have to balance the steps/guidance/order for your intended effect to bring out the best image rather than just bigger number better.

**Review by yiyixuxu — APPROVED** [review]
thanks!

File: docs/source/en/using-diffusers/scheduler_features.md [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+When you take a look at the scheduler's `timesteps` parameter, you'll see that it is the same as the AYS timestep schedule because the `timestep` schedule is calculated from the `sigmas`.
+
+```py
+print(f" timesteps: {pipe.scheduler.timesteps}")
+"timesteps: tensor([999., 845., 730., 587., 443., 310., 193., 116.,  53.,  13.], device='cuda:0')"
+```
+
+### Karras sigmas
+
```
Comment (yiyixuxu) [inline docs/source/en/using-diffusers/scheduler_features.md:152]: maybe add a tip here that @Beinsezii mentioned
that you should not use Karras sigmas on models that weren't trained for it

Comment (yiyixuxu) [reply docs/source/en/using-diffusers/scheduler_features.md:152]: actually, @Beinsezii, can you help provide an example of which model we can't use Karras sigmas for?

**Comment by yiyixuxu** [comment]
@Beinsezii 

for this, if you are able to contribute a doc we would be so grateful! 

>so I think solver order be explored in more depth either here or another doc because the best one is highly dependent on the rest of the params. Like, if you're going run 50 steps anyways a 1st order sampler will have plenty strong enough prediction with less hallucinations. Really have to balance the steps/guidance/order for your intended effect to bring out the best image rather than just bigger number better.

File: docs/source/en/using-diffusers/scheduler_features.md (hunk 1 above)
Comment (Beinsezii) [reply docs/source/en/using-diffusers/scheduler_features.md:152]: SDXL Base shouldn't use karras. It just makes it noisier.

The DreamShaperXL models *are* trained with Karras and look better with it.

**Pull request merged**